health.go

```diff
@@ -229,3 +244,22 @@ func (h *health) updateResult(
 	h.results[name] = result
 	return result
 }
+
+func (h *health) registerShutdownHook(cancel context.CancelFunc) {
+	sigChan := make(chan os.Signal, 1)
+	signal.Notify(sigChan, os.Interrupt, syscall.SIGTERM)
+
+	go func() {
+		for range sigChan {
+			cancel()
+			return
+		}
+	}()
+}
+
+func getContext(parent context.Context, t time.Duration) (context.Context, context.CancelFunc) {
```

Comment:
`getContext` is not very descriptive. I'd use something like `contextWithTimeout` or `ctxWithTimeout`.